it/deployment_ready_test.go

```diff
@@ -0,0 +1,172 @@
+				},
+				Spec: apps_v1.DeploymentSpec{
+					Selector: &meta_v1.LabelSelector{
+						MatchLabels: labelMap,
+					},
+					Replicas: &replicas,
+					Template: core_v1.PodTemplateSpec{
+						ObjectMeta: meta_v1.ObjectMeta{
+							Labels: labelMap,
+						},
+						Spec: core_v1.PodSpec{
+							Containers:                    containers,
+							DNSPolicy:                     "ClusterFirst",
+							RestartPolicy:                 "Always",
+							SchedulerName:                 "default-scheduler",
+							SecurityContext:               &core_v1.PodSecurityContext{},
+							TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+						},
+					},
+					MinReadySeconds:         minReadySeconds,
+					ProgressDeadlineSeconds: &progressDeadlineSeconds,
+					RevisionHistoryLimit:    &revisionHistoryLimit,
+
+					Strategy: apps_v1.DeploymentStrategy{
+						Type: "RollingUpdate",
+						RollingUpdate: &apps_v1.RollingUpdateDeployment{
+							MaxUnavailable: &intstr.IntOrString{
+								Type:   intstr.String,
+								StrVal: "25%",
+							},
+							MaxSurge: &intstr.IntOrString{
+								Type:   intstr.String,
+								StrVal: "25%",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	bundle := &smith_v1.Bundle{
+		TypeMeta: meta_v1.TypeMeta{
+			Kind:       smith_v1.BundleResourceKind,
+			APIVersion: smith_v1.BundleResourceGroupVersion,
+		},
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: "bundle-dt",
+		},
+		Spec: smith_v1.BundleSpec{
+			Resources: []smith_v1.Resource{deployment},
+		},
+	}
+
+	return bundle
+}
+
+func TestDeploymentReady(t *testing.T) {
+	bundle := constructBundle(t, 10, "-l", "0", "-r", "0")
+	SetupApp(t, bundle, false, true, assertSuccess)
+}
+
+func TestDeploymentNeverReady(t *testing.T) {
+	bundle := constructBundle(t, 10, "-l", "20", "-r", "20")
+	SetupApp(t, bundle, false, true, assertDeadlineExceeded)
+}
+
+func assertSuccess(ctx context.Context, t *testing.T, cfg *Config, args ...interface{}) {
+	bundleRes := cfg.AwaitBundleCondition(ctx, IsBundleStatusCond(cfg.Namespace, cfg.Bundle.Name, smith_v1.BundleReady, cond_v1.ConditionTrue))
+
+	smith_testing.AssertCondition(cfg.T, bundleRes, smith_v1.BundleReady, cond_v1.ConditionTrue)
+	smith_testing.AssertCondition(cfg.T, bundleRes, smith_v1.BundleInProgress, cond_v1.ConditionFalse)
+	smith_testing.AssertCondition(cfg.T, bundleRes, smith_v1.BundleError, cond_v1.ConditionFalse)
+}
+
+func assertDeadlineExceeded(ctx context.Context, t *testing.T, cfg *Config, args ...interface{}) {
+	bundleRes := cfg.AwaitBundleCondition(ctx, IsBundleStatusCond(cfg.Namespace, cfg.Bundle.Name, smith_v1.BundleError, cond_v1.ConditionTrue))
+
+	smith_testing.AssertCondition(cfg.T, bundleRes, smith_v1.BundleReady, cond_v1.ConditionFalse)
+	smith_testing.AssertCondition(cfg.T, bundleRes, smith_v1.BundleInProgress, cond_v1.ConditionFalse)
+	smith_testing.AssertCondition(cfg.T, bundleRes, smith_v1.BundleError, cond_v1.ConditionTrue)
```

Comment:
Would be also good to check resource statuses here and in the other test above. To ensure error is reported properly for the resource. See other tests e.g. https://github.com/atlassian/smith/blob/153403692192a45877f4f1da6caf4ea0954a8ab3/it/adoption_test.go#L180-L184

Reply:
Done